### PR TITLE
Fix Item Fresh Brewfest Hops

### PR DESCRIPTION
**Pull request by IntelligentQuantum**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7997
- Closes https://github.com/chromiecraft/chromiecraft/issues/1800

## Tests Performed:

- Tested, Everything works fine except i don't know how can i get speed mount on HandleAuraMounted function
- Arch Linux

## How to Test the Changes:

1. Buy any of brewfest transformation items
2. Mount up
3. Your mount transform to brewfest kodo

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -2832,6 +2833,19 @@ void AuraEffect::HandleAuraMounted(AuraApplication const* aurApp, uint8 mode, bo
                 creatureEntry = 15665;
         }
 
+        // Festive Brewfest Mount
+        if (!GetBase()->HasEffectType(SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED) && target->HasAura(66052))
+        {
+            if (caster->GetSpeedRate(MOVE_RUN) >= 2.0f)
+            {
+                creatureEntry = 27707;
```
Comment (Helias) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:2841]: we should make an enum with all constants of this and remove these magic numbers
like
```c++
enum brewfestEntries = {
  GREAT_BREWFEST_KODO = 27707,
  npc_name_2 = 30507,
  FESTIVE_BREWFEST_MOUNT = 66052,
  ... and so on ...
}
```

**Review by Helias — CHANGES_REQUESTED** [review]
replace the numbers with constant

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 2]
```diff
@@ -5375,6 +5389,34 @@ void AuraEffect::HandleAuraDummy(AuraApplication const* aurApp, uint8 mode, bool
                             }
                         }
                         break;
+                    case 66052: // Festive Brewfest Mount
```
Comment (Helias) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:5392]: you can convert this into a constant and add in the enum below

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 3]
```diff
@@ -5375,6 +5389,34 @@ void AuraEffect::HandleAuraDummy(AuraApplication const* aurApp, uint8 mode, bool
...
+                                {
+                                    creatureEntry = GREAT_BREWFEST_KODO;
+                                }
+                                else
+                                {
+                                    creatureEntry = BREWFEST_KODO;
+                                }
+                            }
+                            else
+                                creatureEntry = target->GetAuraEffectsByType(SPELL_AURA_MOUNTED).front()->GetMiscValue();
```
Comment (Kitzunu) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:5409]: brackets

**Review by Nefertumm — APPROVED** [review]
Related to get mount speed, you can check if the player has X riding skill level (if I understand it properly hehe)

**Review by Nefertumm — APPROVED** [review]
As for the mount speed, you can check riding skill (if I understand you properly hehe)

**Review by Helias — APPROVED** [review]

**Comment by Nefertumm** [comment]
Tested 
image

**Pull request merged**